rpc/src/v1/impls/light/parity.rs

```diff
@@ -277,6 +277,21 @@ impl Parity for ParityClient {
 		)
 	}
 
+	fn all_transaction_hashes(&self) -> Result<Vec<H256>> {
+		let txq = self.light_dispatch.transaction_queue.read();
+		let chain_info = self.light_dispatch.client.chain_info();
+
+		let current = txq.ready_transactions(chain_info.best_block_number, chain_info.best_block_timestamp);
+		let future = txq.future_transactions(chain_info.best_block_number, chain_info.best_block_timestamp);
+		Ok(
+			current
+				.into_iter()
+				.chain(future.into_iter())
+				.map(|tx| Transaction::from_pending(tx).hash.into())
+				.collect::<Vec<_>>()
+		)
+	}
```

Comment:
@seunlanlege also here is copy-paste from `all_transactions`, can this be solved through `all_transactions` or some shared function? or current is ok?

Reply:
you can move the shared logic into a function in `helpers/mod.rs` that returns the chained iterator of `ready_transactions` and `future_transactions`

```rust
/// returns an Iterator of pending transactions, intended for use by light clients
pub fn all_transactions(ligh_dispatch: &Arc<LightDispatch>) -> impl Iterator<Item = PendingTransaction> {
    /// shared logic here, return the chained iterator
}
```

then for `all_transaction_hashes` you can `map` out the hashes, and for `all_transactions` you can `map` into `Transaction::from_pending`

Reply:
what you can do is move the shared logic into `helpers/all_transactions.rs`

basically have a function that returns an iterator of all transactions. 

```rust
pub fn all_transactions(dispatch: &Arc<LightDispatch>) -> impl Iterator<Item=PendingTransaction> {
    // fetch future_transactions and ready_transactions here
    // return the chained iterator of both
}
```

Then you can use `helpers::all_transactions()` depending on the rpc call

for `all_transaction_hashes`

```rust
fn all_transaction_hashes(&self) -> Result<Vec<H256>> {
   // personally i would prefer you access the hash directly on PendingTransaction instead of converting to Transaction first
    helpers::all_transactions(&self.light_dispatch)
        .map(|tx| Transaction::from_pending(tx).hash.into())
        .collect()
}
```

`all_transactions`

```rust
fn all_transactions(&self) -> Result<Vec<Transaction>> {
    helpers::all_transactions(&self.light_dispatch)
        .map(|tx| Transaction::from_pending(tx))
        .collect()
}
```

Reply:
Thanks for examples, this was very helpful. Not sure about helper location, but should be done now.